### PR TITLE
Larger Buffer RingBuffer.h

### PR DESCRIPTION
**Pull request by d3str0y3rport**
The increase of the buffer size to handle larger streams like the ones from GPS.

### CODE REVIEW

**Comment by Adminius** [comment]
why not let user define his own buffer size?

> #ifndef defined(SERIAL_BUFFER_SIZE)
> #define SERIAL_BUFFER_SIZE 64
> #endif

**Comment by tuxedo0801** [comment]
@cmaglie @sandeepmistry Any comments on this?

**Comment by Adminius** [comment]
@d3str0y3rport sorry, i make a mistake:
1. possibility:
`#ifndef SERIAL_BUFFER_SIZE`
2. possibility
`#if !defined(SERIAL_BUFFER_SIZE)`
see also: https://github.com/arduino/ArduinoCore-avr/blob/master/cores/arduino/HardwareSerial.h#L42

File: cores/arduino/RingBuffer.h [hunk 1]
```diff
@@ -27,7 +27,9 @@
 // using a ring buffer (I think), in which head is the index of the location
 // to which to write the next incoming character and tail is the index of the
 // location from which to read.
-#define SERIAL_BUFFER_SIZE 64
+#ifndef defined(SERIAL_BUFFER_SIZE)//allows to define the user the buffer size
+#define SERIAL_BUFFER_SIZE 128//Double the buffer size to get larger streams like from GPS
```
Comment (sandeepmistry) [inline cores/arduino/RingBuffer.h]: I'd be up for bumping this to 256 by default.

The comment about doubling the buffer size should be removed before this pull request can be merged.

**Comment by keestux** [comment]
Can you also show how and where you would define a different value? You'd have to define SERIAL_BUFFER_SIZE in the right place.

And what about if you have an application with multiple modules (one .ino and one or more .cpp)? Each module may or may not include Arduino.h (or Uart.h, or RingBuffer.h). If you don't set the same SERIAL_BUFFER_SIZE everywhere then you're screwed.

**Comment by keestux** [comment]
Another note. Each Uart has 2 ringbuffers. That means the data size increases quite a bit if the variant has 3 or 4 Serial devices.

**Comment by sandeepmistry** [comment]
Closing in favour of: 
https://github.com/arduino/ArduinoCore-samd/commit/da9bfe8cf9aa92dc0e4f480be44fcecd5efe1604

**Comment by sandeepmistry** [comment]
@keestux 

> Can you also show how and where you would define a different value?

This would have to be changed in the boards.txt entry of the variant you are using.

**Pull request closed without merging**